queries/luau/highlights.scm

```diff
@@ -0,0 +1,246 @@
+; Preproc
+
+(hash_bang_line) @preproc
+
+;; Keywords
+
+"return" @keyword.return
+
+[
+ "local"
+ "type"
+ "export"
+] @keyword
+
+(do_statement
+[
+  "do"
+  "end"
+] @keyword)
+
+(while_statement
+[
+  "while"
+  "do"
+  "end"
+] @repeat)
+
+(repeat_statement
+[
+  "repeat"
+  "until"
+] @repeat)
+
+[
+  (break_statement)
+  (continue_statement)
+] @repeat
+
+(if_statement
+[
+  "if"
+  "elseif"
+  "else"
+  "then"
+  "end"
+] @conditional)
+
+(elseif_statement
+[
+  "elseif"
+  "then"
+  "end"
+] @conditional)
+
+(else_statement
+[
+  "else"
+  "end"
+] @conditional)
+
+(for_statement
+[
+  "for"
+  "do"
+  "end"
+] @repeat)
+
+(function_declaration
+[
+  "function"
+  "end"
+] @keyword.function)
+
+(function_definition
+[
+  "function"
+  "end"
+] @keyword.function)
+
+;; Operators
+
+[
+  "and"
+  "not"
+  "or"
+  "in"
+  "typeof"
+] @keyword.operator
+
+[
+  "+"
+  "-"
+  "*"
+  "/"
+  "%"
+  "^"
+  "#"
+  "=="
+  "~="
+  "<="
+  ">="
+  "<"
+  ">"
+  "="
+  "&"
+  "~"
+  "|"
+  "?"
+  "//"
+  ".."
+  "+="
+  "-="
+  "*="
+  "/="
+  "%="
+  "^="
+  "..="
+] @operator
+
+;; Variables
+
+(identifier) @variable
+
+; Types
+
+(type (identifier) @type)
+
+(type (generic_type (identifier) @type))
+
+(builtin_type) @type.builtin
+
+((identifier) @type
+  (#lua-match? @type "^[A-Z]"))
+
+; Typedefs
+
+(type_definition "type" . (type) @type.definition "=")
+
+; Constants
+
+((identifier) @constant
+ (#lua-match? @constant "^[A-Z][A-Z_0-9]+$"))
+
+; Builtins
+
+((identifier) @variable.builtin
+ (#any-of? @variable.builtin "_G" "_VERSION" "debug" "io" "jit" "math" "os" "package" "self" "string" "table" "utf8"))
```

Comment:
Most of these are closer to namespaces rather than variables.
Only "self" is a variable and "_VERSION" is more of a constant.

Reply:
I took it from lua (granted it's not the "same" language though)

Reply:
They are the same in lua.

Reply:
yeah I can agree w changing _VERSION, but do you think the others besides self should be `@namespace` instead? (for lua too)

Reply:
Maybe?